File: src/main/java/com/lynas/security/controller/ProtectedController.java
package com.lynas.security.controller;

import org.springframework.http.ResponseEntity;
import org.springframework.security.access.prepost.PreAuthorize;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestMethod;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("protected")
public class ProtectedController {

    @RequestMapping(method = RequestMethod.GET)
    @PreAuthorize("hasAnyRole('ROLE_ADMIN')")
    public ResponseEntity<?> getDaHoney() {
        return ResponseEntity.ok("{\"success\":true}");
    }

}

File: src/main/kotlin/com/lynas/springsecurityjwt/controller/ProtectedController.kt
package com.lynas.springsecurityjwt.controller

import com.lynas.springsecurityjwt.dto.SuccessResponse
import org.springframework.security.access.prepost.PreAuthorize
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/protected")
class ProtectedController {

    @GetMapping
    @PreAuthorize("hasAnyRole('ROLE_ADMIN')")
    fun hello() = SuccessResponse(true)

}
